cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/Fact.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.metrics.store.timeseries;
+
+import java.util.List;
+
+/**
+ * Represents measure in time with tags assigned to it
+ */
+public final class Fact {
+  private final List<TagValue> tagValues;
+  private final MeasureType measureType;
+  private final String measureName;
+  private final TimeValue timeValue;
+
+  public Fact(List<TagValue> tagValues, MeasureType measureType, String measureName, TimeValue timeValue) {
+    this.tagValues = tagValues;
```

Comment:
Usually I'll do a defensive copy here, unless you are certain that every possible usage of this constructor and the getter method won't modify the list (since it's a public class with public methods, it's hard to govern how it's being used).